mkosi/__init__.py

```diff
@@ -3133,7 +3174,7 @@ def save_uki_components(context: Context) -> tuple[Optional[Path], Optional[str]
     stub = shutil.copy2(stub, context.workspace)
     microcode = build_microcode_initrd(context)
 
-    return stub, kver, kimg, microcode
+    return stub, kver, kimg, [microcode] if microcode else []
```

Comment:
I think i would make sense to have `build_microcode_initrd` return a `list[Path]` now instead of an `Optional[Path]`. Even though it only ever returns an empty list or one with one element, it would make the whole thing more uniform.